server/src/main/kotlin/suwayomi/tachidesk/manga/impl/chapter/ChapterForDownload.kt

```diff
@@ -39,12 +35,11 @@ private class ChapterForDownload(
     suspend fun asDownloadReady(): ChapterDataClass {
         if (isNotCompletelyDownloaded()) {
             markAsNotDownloaded()
-
-            val pageList = fetchPageList()
-
-            updateDatabasePages(pageList)
         }
 
+        val pageList = fetchPageList()
+        updateDatabasePages(pageList)
```

Comment:
making the update page list independent of downloads makes it easier for you to restore downloads from other Tachidesk/Tachiyomi installations.

This may lead to other issues like if the server is disconnected from internet this could result in HTTP exceptions, which the downloader retry should handle properly.

Reply:
The functionality is only useful with cbz downloads, as this already works with normal file downloads. In other words, this code change would be useless if cbz is not in play

Reply:
alright